pennylane/circuit_graph.py

```diff
@@ -588,10 +592,27 @@ def greedy_layers(self):
             if not observables[wire]:
                 observables[wire] = [None]
 
-        return (
-            [operations[wire] for wire in operations],
-            [observables[wire] for wire in observables],
-        )
+        if wire_order is not None:
+            temp_op_grid = OrderedDict()
+            temp_obs_grid = OrderedDict()
+
+            permutation = [
+                self.wires.labels.index(i) for i in wire_order.labels if [i] in self.wires
```

Comment:
(out of scope) wouldn't it be nice to _not_ have to always check if a specific wire is in a Wires object via putting it in a list, and instead just be able to do e.g. `if "a" in self.wires` (instead of `if ["a"] in self.wires`). It took me a second to understand why `i` was encapsulated in a list.

Reply:
Yep, I agree. This is an implementation detail of the Wires class though, so I decided best not to modify it here!